modules/hal_infineon/CMakeLists.txt

```diff
@@ -98,3 +99,17 @@ endif()
 if(CONFIG_BT_PSOC6_BLESS)
   add_subdirectory(bless)
 endif()
+
+if(CONFIG_SOC_FAMILY_INFINEON_PSOC4)
```

Comment:
Add directly the needed cmake commands here, mtb named subdirectories with 4-5 line CMake files makes no sense in the context of Zephyr

Put the content of those inline here when selecting PSOC4, drop any reference to cat/mtb naming possible.